PR-Template Update

rigelrozanski opened the pull request:
Often times I catch myself setting PR's to R4R but forgetting to first read through the github explorer `Files changed` view which it's really good at illuminating basic mistakes. It's a waste of time for reviewers to have to make comments on really obvious basic typo's find/replace errors etc. - I think a simple reminder in the PR-template may go a long way

ValarDragon commented:
I don't think we need another checkbox; i think this could go in the part that isn't displayed. 

(I think our pr template has too many checkboxes atm)

rigelrozanski commented:
Yeah I agree - but I do think this one is important - Maybe we can remove some of the other checkboxes?

cwgoes commented:
I'm not sure they really need to be checkboxes, it's just helpful to have the list of reminders - maybe a bulleted list instead?

rigelrozanski commented:
Nah I like the checkboxes - it's a way to indicate that you've performed something - also it's easy to lose track of what you have or haven't done

ValarDragon commented:
Idk, I'd really prefer this be in a bulleted list. Requiring interactivity here is just kinda inconvenient. I don't want to have to delete this line or write a comment every time, or check the checkbox. Having it somewhere where it needs to be read is all that needs to be achieved (really anywhere near the top of the PR template will achieve that). Checkboxes are good for things we want the reviewer to quickly check off (i.e. did you write tests?), a reviewer can't assert whether or not you re-read your changes.

Same goes for "updating cmd/gaia and /examples", that doesn't apply to like 75% of PR's. (Percentage not actually calculated)

rigelrozanski commented:
Fair but a reviewer could choose to not review your PR yet until the creator has checked the check box to hence not waste their time. 

I agree that `updating cmd/gaia and /examples` usually doesn't apply and we should probably remove it, or more it to bulleted/hidden section

alexanderbez commented:
Did we come to any sort of consensus on this?

rigelrozanski commented:
not really - I think we should keep discussing here and then make all the appropriate changes in this PR to merge? - this PR can cover more than what I had originally proposed obv. (or be totally different)

cwgoes commented:
I would express a soft preference for a bulleted list instead of checkboxes - as a reviewer, I double-check completed checkboxes anyways, so whether or not the checkbox is checked isn't useful information - but not that opinionated.

rigelrozanski commented:
Actually I agree on the reviewer points being bullets - because each reviewer should be checking that independantly - I still see the checkboxes for the submitter as a useful way to communicate the progress of the PR to a reviewer

ValarDragon commented:
Why not just keep it as a bulleted list, and have it say "make sure you completed all of the following. If you didn't and it could potentially be applicable, please include a small note why". 

Though I do think this checkbox vs bulleted list may be a bit of bikeshedding prelaunch, so I'm happy with doing either for now.

melekes commented:
Checkboxes do force me to do some stuff (update changelog, ...), so I wouldn't rule them out. But I would only keep them for must-be-done entries. For other misc, bullet points are fine (https://github.com/cosmos/cosmos-sdk/pull/1929#issuecomment-412174794).

rigelrozanski commented:
Cool - I'm going to update this to what I see as most fit with both checkboxes and bullets

rigelrozanski commented:
Okay - I've updated this to switch to bullets for the assigner section - Also I removed a bunch of the unnecessary verboseness - Also removed "updated code comments bullet" I think that it's just way too obvious and is a subset of "reviewed your code in the github explorer" new bullet point
